src/reducers/example.tsx

```diff
@@ -0,0 +1,54 @@
+import { Reducer } from 'redux';
+
+const TOGGLE_ON = 'TOGGLE_ON';
+const TOGGLE_OFF = 'TOGGLE_OFF';
+
+export interface ExampleState {
+  toggle: 'on' | 'off';
+}
+
+interface ToggleOnAction {
+  type: typeof TOGGLE_ON;
+  payload: {};
+}
+
+export function toggleOn(): ToggleOnAction {
+  return {
+    type: TOGGLE_ON,
+    payload: {},
+  };
+}
+
+interface ToggleOffAction {
```

Comment:
I am curious, why is this an `interface` and not a `type`?

Reply:
I was trying to figure out when one would use a `type` vs an `interface` and cam across quite a bit of inaccurate information. [This post](https://medium.com/@martin_hotell/interface-vs-type-alias-in-typescript-2-7-2a8f1777af4c) seems to provide an up to date description of the issue. It seems to me that for the most part they are actually interchangeable, so it might be best to standardize on always using one in particular unless the specific application requires the other. What do you think?

Reply:
Thanks for the link, Bob. I propose that we use `type` everywhere because of this:

> [declaration merging] doesn’t work with type aliases, because type is an unique type entity ( for both global or module scope )

[This](https://medium.com/@martin_hotell/interface-vs-type-alias-in-typescript-2-7-2a8f1777af4c) is saying that you can merge interfaces but *you cannot merge* types. We probably won't need to merge interfaces so we don't need that feature. The reason I think `type` is superior here is that someone could copy/paste the same interface name by mistake and not get a compilation error:

```js
interface LogInAction {
  type: 'LOG_IN';
  payload: {};
}

// oops, this is a typo maybe caused by copy/paste.
interface LogInAction {
  type: 'LOG_OUT_ACTION';
  payload: {};
}
```

Reply:
👍 for using `type` everywhere. Death to `interface`!

Reply:
I also saw this problem about merging interfaces and was about to suggest the use of types everywhere. Yet, there are probably valid use cases for interfaces and we should maybe learn more about why interfaces are encouraged in TS. Also, it is very unfortunate to have two options for that..